Alignment/OfflineValidation/scripts/haddws.C

```diff
@@ -0,0 +1,304 @@
+ * http://root.cern.ch/drupal/content/downloading-root
+ *
+ * I have added several very useful features to that:
+ * 1. Merge/Add histograms with weights.
+ *		For this, a weight should be specified for each of the input root files.
+ *		Remark: Unfortunately there is no easy way to merger trees with weight
+ *		         so they are just added together (the default behaviour of ROOT)
+ *		         ignoring the specified weights.
+ * 2. Command line execution. No need to load ROOT interpreter every time.
+ *
+ * @Author  : Samantha Hewamanage <samanthaATSPAMMENOTfnal.gov>
+ * @Employer: Florida International University for the CMS Colloboration
+ *****************************************************************************/
+
+/***********************************************************************************
+ * Macro was updated for the purpose of TrackerAlignment group "All-In-One tool"
+ * Changes are mostly of informative character and dealing with inconsistent binning
+ * @Author: Tomas Kello <tomas DOT kello AT cern.ch>
+ ***********************************************************************************/ 
+
+#include <string.h>
+#include "TChain.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TTree.h"
+#include "TKey.h"
+#include "Riostream.h"
+#include <vector>
+#include <utility>
+#include "TProfile.h"
+#include <cstdlib>
+#include <sstream>
+
+using namespace std;
+
+TFile *Target;
+typedef vector<pair<TFile*, double> > vec_pair;
+typedef vector<pair<TFile*, double> >::const_iterator vec_pair_it;
+
+void MergeRootfile( TDirectory *target, const vector<pair<TFile*, double> >& vFileList);
+void loginfo(char log_type, std::string text);
+
+
+void loginfo(char log_type='i', std::string text="") {
+    //**************************************************************************************************************************
+    //Logger:
+    //    INFO    = Informative text
+    //    WARNING = Notify user about unpredictable changes or missing files which do not result in abort
+    //    ERROR   = Error in logic results in abort. Can be fixed by user (missing input, settings clash ...)
+    //    FATAL   = Fatal error results in abort. Cannot be fixed by user (the way how input is produced has changed or bug ...)
+    //**************************************************************************************************************************
+
+    std::string source = "haddws:                    ";
+    if( log_type == 'i') {
+        std::cout << source << "[INFO]     " << text << std::endl; 
+    } else if ( log_type == 'w' ) {
+        std::cout << source << "[WARNING]  " << text << std::endl;
+    } else if ( log_type == 'e' ) {
+        std::cout << source << "[ERROR]    " << text << std::endl;
+    } else if ( log_type == 'f' ) {
+        std::cout << source << "[FATAL]    " << text << std::endl;
+    }
+    return;    
+}
+
+/* entry point for commandline executable */
+int main(int argc, char* argv[])
+{
+
+	if (argc<2)
```

Comment:
oh yeah this file is borrowed and adjusted, I forgot to check indentation

Reply:
> this means that you would have a "command" haddws.C in the prompt, 

why not? `cmssw` provides natively `hadd` via ROOT, but not *`hadd` with weights* (which is what this file does), I can see it could have some general use (then maybe it should be added elsewhere...)

Reply:
Actually, I have just realized it is not a case and scram b will not copy `haddws.C` from `script` directory by default -> `BuildFile.xml` in bin directory will only care for creating haddws executable if source file is already present in bin. Do you know where to change this? Otherwise I can move `haddws.C` directly to bin.

Reply:
> why not?

oh, I have nothing against having an executable :) All the contrary, you should know me :) It's just that, as it is, the script is certainly not going to be compiled, and I thought it would just be copy-pasted in the `bin` directory along with the other executables. I think that originally, `scripts` is intended for executable scripts (like in zsh).

> Actually, I have just realized it is not a case and scram b will not copy

Perhaps `scram b` is smart enough not to copy the non-executable files ? I haven't checked (and it does not matter :) ). If you want to compile it properly like other executables, as you suggest, you should move it to `bin` indeed and adapt the `BuildFile.xml` accordingly.

Reply:
> just that, as it is, the script is certainly not going to be compiled, and I thought it would just be copy-pasted in the bin directory along with the other executables.

That's why (quoting myself):

>  (then maybe it should be added elsewhere...)

I only wanted to make clear that having this as an executable command like `hadd` in native root would be handy for a variety of cases...